graphql/mutation_resolver.go

```diff
@@ -1265,8 +1265,23 @@ func (r *mutationResolver) UpdateUserSettings(ctx context.Context, userSettings
 
 // RemoveItemFromCommitQueue is the resolver for the removeItemFromCommitQueue field.
 func (r *mutationResolver) RemoveItemFromCommitQueue(ctx context.Context, commitQueueID string, issue string) (*string, error) {
-	username := gimlet.GetUser(ctx).DisplayName()
-	result, err := data.FindAndRemoveCommitQueueItem(ctx, commitQueueID, issue, username, fmt.Sprintf("removed by user '%s'", username))
+	user := mustHaveUser(ctx)
+
+	project, err := data.FindProjectById(commitQueueID, false, false)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("could not find project '%s': %s",
+			commitQueueID, err.Error()))
+	}
+
+	if err = data.CheckCanRemoveCommitQueueItem(ctx, r.sc, user, project, issue); err != nil {
```

Comment:
Are we able to check this in a directive the way we check project settings?

Reply:
oops yes... added directive with new tests in which I tested for the `includeRepo` case 😃